invenio_files_rest/cli.py

```diff
@@ -56,15 +59,58 @@ def cp(source, bucket, checksum, key_prefix):
     db.session.commit()
 
 
-@files.command()
+def _unset_default_location():
+    """Unset default location, if there is one."""
+    default_location = Location.get_default()
+    if default_location is not None:
+        default_location.default = False
+
+
+@files.group()
+def location():
+    """Manage locations."""
+
+
+@location.command()
 @click.argument('name')
 @click.argument('uri')
 @click.option('--default', is_flag=True, default=False)
 @with_appcontext
-def location(name, uri, default):
-    """Create new location."""
-    from .models import Location
+def create(name, uri, default):
+    """Create a new location."""
+    if default:
+        _unset_default_location()
     location = Location(name=name, uri=uri, default=default)
     db.session.add(location)
     db.session.commit()
-    click.secho(str(location), fg='green')
+    click.secho('Location {} {} as default {} created'.format(
+        location.name, location.uri, str(location.default)), fg='green')
```

Comment:
Can we make this operation idempotent instead? i.e. make `create` (can rename) ensure that a `Location` with the specified name/uri/default exists (only creates it if it doesn't)? This would make its usage on the command line much smoother: it can be repeatedly run without issue. Right now, this will raise an exception if a Location already exists, even though at the end of the day the goal is just to have this entry in the DB.

Reply:
I see your point. However, since the PK is only the `name` is unique the only editable fields are `uri` and `default`. I only see the use case to change the default, if you need a new `uri` you might as well just create a new location. 

IMHO deleting/removing or even editing a location might cause undesired effects if it already contains files. 

That said I agree on allowing the command be run multiple times without exception (i.e. catching it and showing a red message like "location name already exists".

@fenekku  WDYT?

Reply:
Yes that would work. Something like: if name / uri already exists, just set default if explicit default, else show message (probably yellow) about location already existing. If the name doesn't exit, create.

Reply:
Fixed.

It now shows a yellow message if the name exists. However, following the same logic than above I opted by not setting it default, since for that there is a command (and again it will require explicit an action by the user).